src/main/java/core/basesyntax/model/Operation.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public enum Operation {
+    BALANCE("b"),
+    PURCHASE("p"),
+    SUPPLY("s"),
+    RETURN("r");
+
+    private final String code;
+
+    Operation(String code) {
+        this.code = code;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static Operation getValueFromCode(String code) {
+        return Arrays.stream(Operation.values())
+                .filter(c -> c.getCode().equals(code))
+                .findFirst().orElseThrow();
```

Comment:
When using orElseThrow without an argument, it throws a NoSuchElementException without a message. It's better to provide a custom exception message for clarity. For example: `.orElseThrow(() -> new IllegalArgumentException("No operation found for code: " + code));`